feat: add cookie consent banner

b95702041 opened the pull request:
- Add CookieBanner component with website styling
- Add cookieUtils for consent management and analytics
- Update layout.tsx to integrate CookieBanner

b95702041 committed:
feat: add cookie consent banner

- Add CookieBanner component with website styling
- Add cookieUtils for consent management and analytics
- Add AnalyticsInitializer for default gtag consent state
- Update layout.tsx to integrate CookieBanner

jack482653 commented on components/cookieUtils.ts:
```diff
@@ -0,0 +1,145 @@
+ * Initialize consent on page load
+ * Call this in your layout or app component
+ */
+export const initializeCookieConsent = (): void => {
```
looks like initializeCookieConsent() is unused

b95702041 replied:
updated

jack482653 commented on components/cookieUtils.ts:
```diff
@@ -0,0 +1,145 @@
+
+  // Google Analytics (gtag.js)
+  if (typeof window !== 'undefined' && (window as any).gtag) {
+    (window as any).gtag('consent', 'update', {
```
same as https://github.com/sciwork/swconf2025/pull/11/files#r2441760376

b95702041 commented on components/cookieUtils.ts:
```diff
@@ -0,0 +1,145 @@
+/**
+ * Log current consent status (for debugging)
+ */
+export const logConsentStatus = (): void => {
```
updated

b95702041 commented on components/cookieUtils.ts:
```diff
@@ -0,0 +1,145 @@
+
+  // Google Analytics (gtag.js)
+  if (typeof window !== 'undefined' && (window as any).gtag) {
+    (window as any).gtag('consent', 'update', {
```
I added a new components/AnalyticsInitializer.tsx file to set the default consent state, while keeping the original code to update (enable/disable) analytics.

jack482653 commented:
It still collects data after I reject cookies
<img width="1525" height="492" alt="CleanShot 2025-10-21 at 18 42 16" src="https://github.com/user-attachments/assets/06a9e455-63c6-4775-9b3d-035f0527dd6b" />

b95702041 committed:
refactor: use Button component in CookieBanner and reorganize utils

- Add 'secondary' variant to Button component for gray styling
- Replace hard-coded button elements in CookieBanner with Button component
- Move cookieUtils from components/ to utils/ folder

b95702041 commented on components/cookieUtils.ts:
```diff
@@ -0,0 +1,145 @@
+// components/cookieUtils.ts
```
Moved under new utils folder.

b95702041 committed:
fix: set GA consent default before script loads

- Change AnalyticsInitializer from Client to Server Component
- Use next/script with beforeInteractive strategy
- Ensures consent is denied before any GA requests are sent
- Remove duplicate cookieUtils.ts from components folder
